### ISSUE TITLE
provide option to avoid entering insert mode on BufEnter

### ISSUE THREAD

**Issue by beardedsakimonkey** [gpanders/fennel-repl.nvim#2]
Hi, I personally prefer not having the repl buffer entering insert mode on BufEnter.  Would you be open to having an option to avoid setting up this autocommand?  (e.g. https://github.com/beardedsakimonkey/fennel-repl.nvim/commit/059a7668fc9e98520ae9225f53bb9868f9d31f79)
Note to self: need to default to true and document the default value
Thanks

**Comment by gpanders** [comment]
If we can find a solution to #1, then you can simply delete the BufEnter autocommand.

**Comment by beardedsakimonkey** [comment]
I've been unsuccessful in deleting the autocommand from `after/ftplugin/fennel.fnl` and I think I realized why.  I think it's because the `filetype` is set before installing the autocommand:
https://github.com/gpanders/fennel-repl.nvim/blob/6f91b9449f00a37505f748a63edca7e40691109d/fnl/fennel-repl.fnl#L20
I believe this line triggers my `after/ftplugin/fennel.fnl` to be run, which in my case, looks like this:
```fennel
;; fennel-repl
(when (= :prompt (vim.opt.buftype:get))
  (vim.api.nvim_clear_autocmds {:event :BufEnter :buffer 0}))
```
...and then the plugin continues on to install the autocommand
https://github.com/gpanders/fennel-repl.nvim/blob/6f91b9449f00a37505f748a63edca7e40691109d/fnl/fennel-repl.fnl#L28

Do you foresee any issue with setting the `filetype` *after* configuring the buffer with keymaps and the autocommand such that they can be overridden?   If not, LMK if you'd like me to make PR.

**Comment by gpanders** [comment]
Why not just use

```lua
local buf = require('fennel-repl').open()
vim.api.nvim_clear_autocmds({event = 'BufEnter', buffer = buf})
```

But in any case,

>Do you foresee any issue with setting the filetype after configuring the buffer with keymaps and the autocommand such that they can be overridden? If not, LMK if you'd like me to make PR.

I think this is fine.

**Comment by gpanders** [comment]
Closed in 2befc49ffad39ba81cc449a7a1dd8ea103543dae.

**Comment by beardedsakimonkey** [comment]
Thanks!
> Why not just use

I think that would work, but I'd also like to remove the autocommand when opening the repl via `:FennelRepl`
